### PR TITLE
7815 remove unused static variables in libresolve2

### PR DESCRIPTION
**Pull request by ikozhukhov**
cleanup of unused static variables.
have been found my gcc6

### CODE REVIEW

**Comment by prakashsurya** [comment]
@zettabot go

**Comment by jclulow** [comment]
This change doesn't seem to be ZFS-related at all.  If you're looking to get this integrated into `illumos-gate`, you'll still need to go through the usual review, build and test process in an appropriate venue before seeking integration.

**Comment by ikozhukhov** [comment]
really, i don’t understand - what are issues here with my changes?

you are, as Joyent worker, no contributing to illumos.
I see only Robert’s updates, who do it for all Joyent persons.

what the reason do not use OpenZFS env for builds and testing?
it's mean - my corrent changes will be built on illumos supported env and tested by zfs tests.

yes - they are changes not related to ZFS, but, after build, they will be tested by loades to test env where libresolve2 is using.

i have forwarded this PR to illumos dev maillist for review - this step needs for illumos contribution.

BUT, you can rejecte it for illumos parts, but it can be interest and integrate to OpenZFS community.

i have DilOS where i do builds and testing and i wanted to contribute these updates, because Richard Lowe asted about it on IRC: upstream it - for my gcc6 updates.

if it is NOT applicable for illumos tree - just say about it: do not integrate it.
maybe OpenZFS tree will be interested in cleanups of issues what can be found by another compiler.

I can say again - i have tested my changes on my DilOS - and try contribute it.

I don’t want waste my time for OpenIndiana separate setup and maintain it.

I want to hear resolution: can I contribute changes based on DilOS or NOT.
it will save my time.

best regards,
-Igor

**Comment by ahrens** [comment]
@jclulow 
> This change doesn't seem to be ZFS-related at all.

I agree.

> If you're looking to get this integrated into illumos-gate, you'll still need to go through the usual review, build and test process in an appropriate venue before seeking integration.

Review - I agree that there needs to be an opportunity to review this code on developer@lists.illumos.org.  There should be an email there shortly inviting review.  I think that github (on either illumos's or OpenZFS's github account) is an acceptable tool for giving and responding to code review feedback, as is ReviewBoard, as is email to developer@lists.illumos.org.

Build - I agree that the submitted code needs to be built.  The infrastructure here (the OpenZFS github account) will do builds based on the OpenZFS repo and OmniOS.  This repo is nearly identical to illumos-gate (IIRC, the only difference is that it may be a few days behind, and the README.md).  If this is sufficient building for ZFS-related changes, I think it should be sufficient for non-ZFS changes as well.

Test - I agree that all changes to illumos need to be tested, and running the zfs test suite is typically insufficient testing for non-ZFS changes.  The criteria of "usual and appropriate" testing should be discussed on developer@lists.illumos.org and/or with advocates@ (before or during the RTI process).

In conclusion, I agree with @jclulow.  In my opinion, it may be "unusual" to post a pull request here for non-ZFS changes, but doing so does not preclude proper following of the procedures.  If there's a specific deficiency to this or other changes, please let us know what it is.

**Comment by pyhalov** [comment]
hm.. Files changed shows 85.
git grep -w rcsid usr/src/lib/libresolv2/|cut -d : -f 1|sort |uniq |wc -l 
shows 100. 
Why there's the difference?
